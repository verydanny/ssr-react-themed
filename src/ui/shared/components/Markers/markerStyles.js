import {
  PinIcon,
  SelectedPinIcon,
  InactivePinIcon,
  PdpPinLogoIcon,
} from './icons'

const BIG_PIN = {
  url: PinIcon,
  anchor: { x: 8, y: 8 },
  scaledSize: { width: 16, height: 16 },
}

const SMALL_PIN = {
  url: PinIcon,
  anchor: { x: 5, y: 5 },
  scaledSize: { width: 11, height: 11 },
}

const INACTIVE_PIN = {
  url: InactivePinIcon,
  anchor: { x: 5, y: 5 },
  scaledSize: { width: 11, height: 11 },
}

const SELECTED_BIG_PIN = {
  url: SelectedPinIcon,
  anchor: { x: 8, y: 8 },
  scaledSize: { width: 16, height: 16 },
}

const SELECTED_SMALL_PIN = {
  url: SelectedPinIcon,
  anchor: { x: 5, y: 5 },
  scaledSize: { width: 11, height: 11 },
}

export const PDP_LOGO_PIN = {
  url: PdpPinLogoIcon,
  anchor: { x: 28, y: 32 },
  scaledSize: { width: 56, height: 64 },
}

export const markerNormalIcon = ({ isActive, isPaid }) => {
  if (!isActive) return INACTIVE_PIN
  return isPaid ? BIG_PIN : SMALL_PIN
}

export const markerSelectedIcon = isPaid => (isPaid ? SELECTED_BIG_PIN : SELECTED_SMALL_PIN)
