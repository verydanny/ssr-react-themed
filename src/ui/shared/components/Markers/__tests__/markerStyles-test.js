import { markerNormalIcon, markerSelectedIcon } from '../markerStyles'

const paidIcon = {
  isPaid: true,
  isActive: true,
}

const unpaidIcon = {
  isPaid: false,
  isActive: true,
}

const inactiveIcon = {
  isPaid: false,
  isActive: false,
}

describe('ui/shared/components/Markers/markerStyles', () => {
  describe('markerNormalIcon', () => {
    it('returns the paid normal icon', () => {
      expect(markerNormalIcon(paidIcon)).toMatchSnapshot()
    })

    it('returns the unpaid normal icon', () => {
      expect(markerNormalIcon(unpaidIcon)).toMatchSnapshot()
    })

    it('returns the inactive normal icon', () => {
      expect(markerNormalIcon(inactiveIcon)).toMatchSnapshot()
    })
  })

  describe('markerSelectedIcon', () => {
    it('returns the active paid selected icon when isPaid is true', () => {
      expect(markerSelectedIcon(true)).toMatchSnapshot()
    })

    it('returns the active unpaid selected icon when isPaid is false', () => {
      expect(markerSelectedIcon(false)).toMatchSnapshot()
    })

    it('returns the inactive selected icon when isPaid is false', () => {
      expect(markerSelectedIcon(false)).toMatchSnapshot()
    })
  })
})
