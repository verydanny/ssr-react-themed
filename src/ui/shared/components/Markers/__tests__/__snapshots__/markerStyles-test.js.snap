// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/shared/components/Markers/markerStyles markerNormalIcon returns the inactive normal icon 1`] = `
Object {
  "anchor": Object {
    "x": 5,
    "y": 5,
  },
  "scaledSize": Object {
    "height": 11,
    "width": 11,
  },
  "url": "",
}
`;

exports[`ui/shared/components/Markers/markerStyles markerNormalIcon returns the paid normal icon 1`] = `
Object {
  "anchor": Object {
    "x": 8,
    "y": 8,
  },
  "scaledSize": Object {
    "height": 16,
    "width": 16,
  },
  "url": "",
}
`;

exports[`ui/shared/components/Markers/markerStyles markerNormalIcon returns the unpaid normal icon 1`] = `
Object {
  "anchor": Object {
    "x": 5,
    "y": 5,
  },
  "scaledSize": Object {
    "height": 11,
    "width": 11,
  },
  "url": "",
}
`;

exports[`ui/shared/components/Markers/markerStyles markerSelectedIcon returns the active paid selected icon when isPaid is true 1`] = `
Object {
  "anchor": Object {
    "x": 8,
    "y": 8,
  },
  "scaledSize": Object {
    "height": 16,
    "width": 16,
  },
  "url": "",
}
`;

exports[`ui/shared/components/Markers/markerStyles markerSelectedIcon returns the active unpaid selected icon when isPaid is false 1`] = `
Object {
  "anchor": Object {
    "x": 5,
    "y": 5,
  },
  "scaledSize": Object {
    "height": 11,
    "width": 11,
  },
  "url": "",
}
`;

exports[`ui/shared/components/Markers/markerStyles markerSelectedIcon returns the inactive selected icon when isPaid is false 1`] = `
Object {
  "anchor": Object {
    "x": 5,
    "y": 5,
  },
  "scaledSize": Object {
    "height": 11,
    "width": 11,
  },
  "url": "",
}
`;
